com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/BuiltInExtensionsTests.cs

```diff
@@ -0,0 +1,234 @@
+            Assert.AreEqual("Path/To/Directory", altPath.SetSeparatorChar());
+        }
+
+        static IEnumerable<TestCaseData> DoubleArrayData
+        {
+            get
+            {
+                yield return new TestCaseData(new double[] {1,1,1}, new double[] {1,1,1,1}).Returns(false);
+                yield return new TestCaseData(new double[] {1,1,1}, new double[] {1,1,1}).Returns(true);
+                yield return new TestCaseData(new double[] {0,0,0}, new double[] {1,1,1}).Returns(false);
+            }
+        }
+
+        [Test, TestCaseSource("DoubleArrayData")]
+        public bool DoubleDeltaCompare_DeltaE4_Success(double[] array, double[] array2)
+        {
+            return array.DoubleDeltaCompare(array2, 1e-4);
+        }
+
+        static IEnumerable<TestCaseData> Vector3Data
+        {
+            get
+            {
+                yield return new TestCaseData(Vector3.zero, Vector3.one).Returns(false);
+                yield return new TestCaseData(Vector3.one, Vector3.one).Returns(true);
+            }
+        }    
+
+        [Test, TestCaseSource("Vector3Data")]
+        public bool VectorEqualDelta_DeltaE4_Success(Vector3 source, Vector3 exported)
+        {
+            return source.VectorEqualDelta(exported, 1e-4);
+        }
+
+        static IEnumerable<TestCaseData> DoubleData
+        {
+            get
+            {
+                yield return new TestCaseData(0, 0, 1e-4).Returns(true);
+                yield return new TestCaseData(0, 1, 1e-4).Returns(false);
+                yield return new TestCaseData(0, 0.001, 1e-4).Returns(false);
+                yield return new TestCaseData(0, 0.0001, 1e-2).Returns(true);
+            }
+        }  
+
+        [Test, TestCaseSource("DoubleData")]
+        public bool EqualsDelta_ModifiedDelta_Success(double first, double second, double delta)
+        {
+            return first.EqualsDelta(second, delta);
+        }
+
+        static IEnumerable<TestCaseData> Matrix4x4SubtractData
+        {
+            get
+            {
+                yield return new TestCaseData(Matrix4x4.identity, Matrix4x4.identity).Returns(Matrix4x4.zero);
+                yield return new TestCaseData(Matrix4x4.zero, Matrix4x4.zero).Returns(Matrix4x4.zero);
+            }
+        }  
+
+        [Test, TestCaseSource("Matrix4x4SubtractData")]
+        public Matrix4x4 Subtract_Matrix4x4s_Success(Matrix4x4 first, Matrix4x4 second)
+        {
+            return first.Subtract(second);
+        }
+
+        [Test]
+        public void FloatDivide_Identities_Success()
+        {
+            Assert.AreEqual(Matrix4x4.identity, Matrix4x4.identity.FloatDivide(1));
+            Assert.AreEqual(Matrix4x4.zero, Matrix4x4.zero.FloatDivide(1));
+        }
+
+        [Test]
+        public void FirstChildByQuery_Components_ReturnFirst()
+        {
+            var parent = new GameObject("Parent").transform;
```

Comment:
destroy the object at the end of the test?

Reply:
ah i knew i'd forget some, thanks for catching!